.github/workflows/go.yml

```diff
@@ -0,0 +1,38 @@
+name: Go
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v ./...
```

Comment:
use mage, e.g., `mage test` and same for build too

Reply:
Done to test, but I kept the go build to build in multi-OS.
WDYT?